### PR TITLE
Reduce the number of string allocations

### PR DESCRIPTION
**Pull request by amorde**
cc @segiddins

### CODE REVIEW

File: lib/xcodeproj/project/object_attributes.rb [hunk 1]
```diff
@@ -354,14 +354,15 @@ def has_one(singular_name, isas)
             # 1.9.2 fix, see https://github.com/CocoaPods/Xcodeproj/issues/40.
             public(attrb.name)
 
+            variable_name = "@#{attrb.name}"
```
Comment (amorde) [inline lib/xcodeproj/project/object_attributes.rb]: The attribute method definitions were allocating over 360k instances of "@dependencies" and "dependencies", so I thought this was worth it :)

File: lib/xcodeproj/project.rb [hunk 2]
```diff
@@ -64,6 +68,7 @@ class Project
     #
     def initialize(path, skip_initialization = false, object_version = Constants::DEFAULT_OBJECT_VERSION)
       @path = Pathname.new(path).expand_path
+      @project_dir = @path.dirname
```
Comment (amorde) [inline lib/xcodeproj/project.rb]: Cuts down on some of the allocations of the `Pods/` directory

File: lib/xcodeproj/project/object_attributes.rb (hunk 1 above)
Comment (segiddins) [reply lib/xcodeproj/project/object_attributes.rb]: nit: `variable_name = :"@#{attrb.name}"` so it's a symbol instead of a string

Comment (amorde) [reply lib/xcodeproj/project/object_attributes.rb]: fixed

**Pull request merged**